2-classes-db/task.md

```diff
@@ -1,79 +1,114 @@
 ```
+
 ### 5. Show the date, transaction_no, supplier_inv_no, description and amount for those transactions whose amount is £25,000, £30,000, £35,000 or £40,000.
-```sql
 
+```sql
+SELECT date, transaction_no, supplier_inv_no, description, amount FROM spends WHERE amount IN (25000,30000,35000,40000);
 ```
+
 ### 6. Show the date, the supplier_id, the description and the amount for transactions with the expense area of 'Better Hospital Food'. You could do a query to get the expense_area_id first then do a query to find the dates, supplier_ids and amounts. But it would be better to do this all in one query by linking the tables together using INNER JOINs.
-```sql
 
+```sql
+SELECT spends.date, spends.supplier_id, spends .description, spends .amount FROM spends  INNER JOIN expense_areas  ON spends.expense_area_id = expense_areas.id  WHERE expense_area = 'Better Hospital Food';
 ```
+
 ### 7. Show the date, supplier name, description and amount for transactions with the expense area of 'Better Hospital Food'. You will need to INNER JOIN another table to be able to do this.
-```sql
 
+```sql
+SELECT date, supplier, description, amount
+FROM spends
+INNER JOIN suppliers ON spends.supplier_id = suppliers.id
+INNER JOIN expense_areas ON spends.expense_area_id = expense_areas.id
+WHERE expense_area='Better Hospital Food';
 ```
+
 ### 8. We have just received a late invoice for April! Add a new row to the spends table:
+
     dated 1st April 2021
     with a description of 'Computer Hardware Dell'
     transaction number is 38104091 and the supplier's inv no is '3780119655'
     the supplier is 'COMPUTACENTER (UK) LTD' (id 16)
     the expense type is 'Computer Hardware Purch' (id 7)
     the expense area is 'ICT Contingency' (id 18)
     for £32,000.
-```sql
 
+```sql
+INSERT INTO spends (expense_type_id, expense_area_id, supplier_id,date,transaction_no,supplier_inv_no,description, amount)
+VALUES(7,18,16, '2021-04-01',38104091,3780119655, 'Computer Hardware Dell', 32000);
 ```
+
 ### 9. If you examine the dates in the data, you will see they all are dated either 1st march 2021 or 1st April 2021. So if we group on the the date, there will only be two groups. Show the date and the total amount spent on that date for these two dates by using a GROUP BY clause.
-```sql
 
+```sql
+SELECT date, SUM(amount) as total_amount_spent
+FROM spends GROUP BY date;
 ```
+
 ### 10. (optional) Great we now know the monthly spend. But it didn't look that good. So I've changed my SELECT query to output this instead:
+
 ```
-   Month    | Monthly Spend 
+   Month    | Monthly Spend
 ------------+---------------
  March 2021 | £ 28,674,452
  April 2021 | £ 22,895,194
 (2 rows)
 ```
+
 Can you work out how to do this?
 
 ```sql
-
+SELECT TO_CHAR(date, 'Month YYYY') AS "Month", CONCAT('£ ', TO_CHAR(SUM(amount), '999,999,999')) AS "Monthly Spend"
+FROM spends
+GROUP BY date;
```

Comment:
CORRECT - Note that it would be possible to simplify the currency formatting, rather than doing it long-hand in this way (see https://www.techonthenet.com/postgresql/functions/to_char.php)